fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -2074,4 +2098,203 @@ private void populateCodesSubSegment(StringBuilder whereClauseSegment, Modifier
         log.exiting(CLASSNAME, METHODNAME);
     }
 
+    private SqlQueryData processProfileParm(Class<?> resourceType, QueryParameter queryParm, String tableAlias)
+            throws FHIRPersistenceException {
+        final String METHODNAME = "processProfileParm";
+        log.entering(CLASSNAME, METHODNAME, queryParm.toString());
+
+        // Join to the canonical parameter table...which in this case means the xx_profiles table
+        // because currently _profile is the only search parameter to be defined as a canonical
+
+        StringBuilder whereClauseSegment = new StringBuilder();
+        boolean parmValueProcessed = false;
+        SqlQueryData queryData;
+        List<Object> bindVariables = new ArrayList<>();
+
+        whereClauseSegment.append(LEFT_PAREN);
+        for (QueryParameterValue value : queryParm.getValues()) {
+            // If multiple values are present, we need to OR them together.
+            if (parmValueProcessed) {
+                whereClauseSegment.append(OR);
+            }
+
+            // Reuse the same CanonicalSupport code used for param extraction to parse the search value
+            ResourceProfileRec rpc = CanonicalSupport.makeResourceProfileRec(-1, resourceType.getSimpleName(), -1, -1,
+                    value.getValueString(), false);
+            int canonicalId = identityCache.getCanonicalId(rpc.getCanonicalValue());
+            whereClauseSegment.append(tableAlias).append(DOT).append("CANONICAL_ID").append(EQ).append(canonicalId);
+            
+            // TODO double-check semantics of ABOVE and BELOW in this context
```

Comment:
Is this TODO still needed?

Reply:
I think it needs further discussion. I pulled this code over from some existing NewQueryBuilder code, so same issue there. The spec isn't real clear on the semantics of :above and :below for canonical values.